src/audio/copier/copier_dai.c

```diff
@@ -281,3 +281,29 @@ void copier_dai_free(struct copier_data *cd)
 		rfree(cd->dd[i]);
 	}
 }
+
+int copier_dai_prepare(struct comp_dev *dev, struct copier_data *cd)
+{
+	int ret = 0;
+
+	if (cd->endpoint_num == 1) {
+		ret = dai_zephyr_config_prepare(cd->dd[0], dev);
+		if (ret < 0)
+			return ret;
+
+		ret = dai_zephyr_prepare(cd->dd[0], dev);
+		if (ret < 0)
+			return ret;
+	} else {
+		for (int i = 0; i < cd->endpoint_num; i++) {
+			ret = dai_zephyr_config_prepare(cd->dd[i], dev);
+			if (ret < 0)
+				return ret;
+
+			ret = dai_zephyr_prepare(cd->dd[i], dev);
+			if (ret < 0)
+				return ret;
+		}
+	}
+	return ret;
```

Comment:
is this function expected to return positive values? If not, just `return 0` here and remove initialisation